### PR TITLE
Add proxy to operator deploy

### PR DESCRIPTION
**Pull request by mikepfaffroth-okta**
#### What this PR does / why we need it:
This allows setting of the necessary variables to allow the operator to use a proxy for outbound connections.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [X] Chart Version bumped
- [X] `CHANGELOG.md` has been updated
- [X] Variables are documented in the `README.md`

### CODE REVIEW

**Review by maycmlee — APPROVED** [review]
👍  for docs.

File: charts/datadog-operator/values.yaml [hunk 1]
```diff
@@ -105,6 +105,14 @@ podAnnotations: {}
...
 podLabels: {}
 
+proxy:
+  # proxy.http -- Sets a proxy for all HTTP connections.
+  http: ""
+  # proxy.https -- Sets a proxy for all HTTPS connections.
+  https: ""
+  # proxy.none -- Sets networks to send traffic without a proxy.
+  none: ""
+
```
Comment (clamoriniere) [inline charts/datadog-operator/values.yaml:115]: The solution that you propose here is valid, but on the `datadog` chart, we used another approach.

To support proxy configuration threw environment variable configuration: we provide a way to configure any env-vars https://github.com/DataDog/helm-charts/blob/e167c72f8f22fcc3a0b3698690a6754e366a00cc/charts/datadog/values.yaml#L319-L324
it give a bit more flexibility.

File: charts/datadog-operator/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: datadog-operator
-version: 0.8.0
-appVersion: 0.8.0
+version: 0.8.1
+appVersion: 0.8.1
```
Comment (clamoriniere) [inline charts/datadog-operator/Chart.yaml:4]:
```suggestion
appVersion: 0.8.0
```

**Review by clamoriniere — COMMENTED** [review]
I added some comments.

**Pull request merged**